api/v1alpha1/shared_types.go

```diff
@@ -574,6 +574,22 @@ type ClusterSettings struct {
 	// +optional
 	Timeout *Timeout `json:"timeout,omitempty"`
 
+	// BufferRequests allows the gateway to buffer and fully receive each request from a client before continuing to send the request
+	// upstream to the backends. This can be helpful to shield your backend servers from slow clients, and also to enforce a maximum size per request
+	// as any requests larger than the buffer size will be rejected.
+	//
+	// This can have a negative performance impact so should only be enabled when necessary.
+	//
+	// When enabling this option, you should also configure your connection buffer size to account for these request buffers. There will also be an
+	// increase in memory usage for Envoy that should be accounted for in your deployment settings.
+	//
+	// When the target of this policy is a gateway the buffer will be applied at a listener level. When the target of the policy is a HTTPRoute it
+	// will be applied for that particular route only.
+	//
+	// +notImplementedHide
+	// +optional
+	BufferRequests *BufferRequests `json:"bufferRequests,omitempty"`
```

Comment:
I think this should be put in `BackendTrafficPolicySpec` instead of `ClusterSettings`.

`BufferRequests` shouldn't be required for non-route backends such as ext aut, ext proc, oidc provider, OTL sink, etc.

Reply:
Ah that was my intention and saw ClusterSettings is inline, but if it's used for other purposes I'll move it directly on to `BackendTrafficPolicySpec` 👍

Reply:
Some features are put in the `ClusterSettings` because they can also be used for non-route backends, like  ext aut, ext proc, OIDC provider, or OTL sink.

BufferRequests can be used for non-route backends. Let's put it directly in the BackendTrafficPolicySpec.